pyproject.toml

```diff
@@ -17,18 +17,33 @@
 [build-system]
 build-backend = 'mesonpy'
 requires = [
-    "meson-python>=0.15.0",
-    "Cython>=3.0.8",        # when updating version, also update check in meson.build
-    "pybind11>=2.13.2",     # when updating version, also update check in scipy/meson.build
-    "pythran>=0.14.0",
+    # The upper bound on meson-python is pre-emptive only (looser
+    # on purpose, since chance of breakage in 0.18/0.19 is low with
+    # 0.17.1 working at time of writing)
+    "meson-python>=0.15.0,<0.20.0",
+    # we need at least Cython 3.1.0a1 for free-threaded CPython
+    # 3.13 wheel builds
+    "Cython>=3.1.0a1,<3.2.0; python_version=='3.13t'",        # when updating version, also update check in meson.build
+    # for other CPython versions, the regular pre-emptive
+    # Cython version bounds policy applies
+    "Cython>=3.0.8,<3.1.0; python_version!='3.13t'",
```

Comment:
same here... I think we'll just have to do the usual thing and disable build isolation, but it is not a very nice solution--having access to `sysconfig.get_config_var("Py_GIL_DISABLED")` might do the track, but don't think we have access to it?

Reply:
No we don't. There is no way to check for free-threading in `pyproject.toml`.